userspace/libsinsp/test/plugins.ut.cpp

```diff
@@ -378,6 +378,51 @@ TEST_F(sinsp_with_test_input, plugin_custom_source) {
 	ASSERT_EQ(next_event(), nullptr);  // EOF is expected
 }
 
+class plugin_test_event_processor : public libsinsp::event_processor {
+public:
+	plugin_test_event_processor() { num_async_evts = 0; }
+
+	void on_capture_start() {}
+
+	virtual void process_event(sinsp_evt* evt, libsinsp::event_return rc) {
+		if(evt->get_type() == PPME_ASYNCEVENT_E) {
+			num_async_evts++;
+		}
+	}
+
+	int num_async_evts;
+};
+
+// scenario: a plugin with dump capability is requested a dump and then the capture file is read.
+TEST_F(sinsp_with_test_input, plugin_dump) {
```

Comment:
Very simple test:
* register a plugin with async event capability
* open inspector in no driver mode
* request a scap file dump to a temporary text file
* at this stage, the plugin will be requested to dump its state; in our test case, the plugin will just dump 10 fake events
* open a replay inspector to read the generated scap file
* register our event processor that just counts number of `PPME_ASYNC_EVENT_E`
* remove the test scap file

Reply:
Why writing this in a GH comment instead of leaving it as a commented note in the code itself? 😂